src/main/java/org/opensrp/web/rest/UploadController.java

```diff
@@ -0,0 +1,378 @@
+			@RequestParam("file") MultipartFile file, HttpServletRequest request) {
+		String teamID = getStringFilter(TEAM_ID, request);
+		String teamName = getStringFilter(TEAM_NAME, request);
+		String locationID = getStringFilter(LOCATION_ID, request);
+
+		String entityId = UUID.randomUUID().toString();
+		String providerId = getCurrentUser();
+		MultimediaDTO multimediaDTO = new MultimediaDTO(entityId.trim(), providerId, file.getContentType().trim(), null,
+				FILE_CATEGORY);
+
+		String status = null;
+		try {
+			List<Map<String, String>> csv_clients = readCSVFile(file);
+
+			UploadValidationBean validationBean = uploadService.validateFieldValues(csv_clients, eventName, globalID);
+			if (validationBean.getErrors() != null && validationBean.getErrors().size() > 0) {
+				validationBean.setAnalyzedData(null);
+				Map<String, Object> map = new HashMap<>();
+				map.put("status", "A number of errors were found during validation");
+				map.put("summary", validationBean);
+				return new ResponseEntity<>(gson.toJson(map), HttpStatus.BAD_REQUEST);
+			}
+
+			UniqueIDProvider uniqueIDProvider = new OpenMRSUniqueIDProvider(openmrsIDService,
+					validationBean.getRowsToCreate());
+
+			for (Pair<Client, Event> eventClient : validationBean.getAnalyzedData()) {
+				Client client = eventClient.getLeft();
+
+				String baseEntityID = client.getBaseEntityId();
+				boolean newClient = false;
+				if (StringUtils.isBlank(baseEntityID)) {
+					baseEntityID = UUID.randomUUID().toString();
+					client.setBaseEntityId(baseEntityID);
+					newClient = true;
+				}
+
+				client.addAttribute(DEFAULT_RESIDENCE, locationID);
+
+				assignClientUniqueID(client, globalID, uniqueIDProvider);
+				clientService.addorUpdate(client);
+
+				// update event details
+				Event event = (eventClient.getRight() == null) ? new Event() : eventClient.getRight();
+				String eventType = newClient ? eventName : "Update " + eventName;
+				prepareEvent(event, baseEntityID, providerId, eventType);
+
+				if (StringUtils.isNotBlank(teamID))
+					event.setTeamId(teamID);
+
+				if (StringUtils.isNotBlank(teamName))
+					event.setTeam(teamName);
+
+				if (StringUtils.isNotBlank(locationID))
+					event.setLocationId(locationID);
+
+				// save the event
+				eventService.addorUpdateEvent(event);
+			}
+
+			Map<String, String> details = new HashMap<>();
+			details.put("size", Long.toString(file.getSize()));
+			details.put("imported", Integer.toString(validationBean.getRowsToCreate()));
+			details.put("updated", Integer.toString(validationBean.getRowsToUpdate()));
+			multimediaDTO.withOriginalFileName(file.getOriginalFilename())
+					.withDateUploaded(new Date())
+					.withSummary(mapper.writeValueAsString(details));
+
+			status = multimediaService.saveFile(multimediaDTO, file.getBytes(), file.getOriginalFilename());
+
+			return new ResponseEntity<>(gson.toJson(details), HttpStatus.OK);
+		}
+		catch (Exception e) {
```

Comment:
Don't catch general exceptions, this should be caught by global exception handler